frontend/src/App.vue

```diff
@@ -54,16 +53,67 @@ export default {
       );
     },
   },
-  watch: {
-  },
+  watch: {},
   methods: {
     async onAction(contextualItem, index) {
       const itemFqid = contextualItem?.item?.fqid;
-      await this.rpc.invoke("performAction", [itemFqid, index, contextualItem.context]);
+      await this.rpc.invoke("performAction", [
+        itemFqid,
+        index,
+        contextualItem.context,
+      ]);
     },
+  getItem(itemFqid) {
+    for (const collection of this.collections) {
+      for (const item of collection.items) {
+        if (item.fqid === itemFqid) {
+          return item;
+        }
+        if (item.items) {
+          for (const subItem of item.items) {
+            if (subItem.fqid === itemFqid) {
+              return subItem;
+            }  
+            else if( subItem.items ){
+              const deepSubitem =  this.getDeepSubitem(itemFqid, subItem);
+              if(deepSubitem){
+                return deepSubitem;
+              }
+            }
+          }
+        }
+      }
+    }
+    // TODO - console log: item does not exist
+  },
+ getDeepSubitem( itemFqId, item ){
+    if( item.items ){
+      for (const subItem of item.items) {
+        if (subItem.fqid === itemFqId) {
+          return subItem;
+        } else if(subItem.items){
+            return this.getDeepSubitem(itemFqId, subItem);
+        } 
+      }
+    }
+  },
+  onClickPanel() {
+    //to do: change state
+ },
     async showCollections(collections) {
       this.collections = collections;
     },
+    changeItemsState(changedItems) {
+      for(let i=0; i < changedItems.length; i++){
```

Comment:
use a `forEach` or `for of` loop instead of instead of `for`
as you don't really need the index `i` here and this will resolve formatting inconsistencies in line 107.